src/connector/src/source/nats/source/reader.rs

```diff
@@ -46,19 +47,59 @@ impl SplitReader for NatsSplitReader {
     ) -> Result<Self> {
         // TODO: to simplify the logic, return 1 split for first version
         assert!(splits.len() == 1);
-        let splits = splits
-            .into_iter()
-            .map(|split| split.into_nats().unwrap())
-            .collect::<Vec<NatsSplit>>();
+        let split = splits.into_iter().next().unwrap().into_nats().unwrap();
+        let start_position = match &split.start_sequence {
+            NatsOffset::None => match &properties.scan_startup_mode {
+                None => NatsOffset::Earliest,
+                Some(mode) => match mode.as_str() {
+                    "latest" => NatsOffset::Latest,
+                    "earliest" => NatsOffset::Earliest,
+                    "timestamp_millis" => {
+                        if let Some(time) = &properties.start_time {
+                            NatsOffset::Timestamp(time.parse()?)
+                        } else {
+                            return Err(anyhow!("scan_startup_timestamp_millis is required"));
+                        }
+                    }
+                    "sequence_number" => {
+                        if let Some(seq) = &properties.start_sequence {
+                            NatsOffset::SequenceNumber(seq.clone())
+                        } else {
+                            return Err(anyhow!("scan_startup_sequence_number is required"));
+                        }
+                    }
+                    _ => {
+                        return Err(anyhow!(
+                            "invalid scan_startup_mode, accept earliest/latest/sequence_number/time"
+                        ))
+                    }
+                },
+            },
+            start_position => start_position.to_owned(),
+        };
+
+        if !matches!(start_position, NatsOffset::SequenceNumber(_))
+            && properties.start_sequence.is_some()
+        {
+            return Err(
+                anyhow!("scan.startup.mode need to be set to 'sequence_number' if you want to start with a specific sequence number")
+            );
+        }
+        if !matches!(start_position, NatsOffset::Timestamp(_)) && properties.start_time.is_some() {
+            return Err(
+                anyhow!("scan.startup.mode need to be set to 'timestamp_millis' if you want to start with a specific timestamp millis")
+            );
+        }
         let consumer = properties
             .common
-            .build_consumer(0, splits[0].start_sequence)
+            .build_consumer(0, start_position.clone())
```

Comment:
I think it is a bad practice to hardcode the split_id `0` everywhere in the code.  Here you should get the split id from the input split. And I suggest that you can save the split id to the `SplitReader` when you create the reader. Then you can use the split_id to build a `SourceMessage` instead of hardcoding it in the `from_nats_jetstream_message`. 
Btw you may `impl From<NatsMessage> for SourceMessage` to do the conversion.

Reply:
Thanks for you remind. I already applied your suggestions. :)